lib/srv/alpnproxy/local_proxy.go

```diff
@@ -316,27 +318,43 @@ func (l *LocalProxy) StartHTTPAccessProxy(ctx context.Context) error {
 	l.cfg.Log.Info("Starting HTTP access proxy")
 	defer l.cfg.Log.Info("HTTP access proxy stopped")
 	defaultProxy := l.makeHTTPReverseProxy(l.getCerts())
-	err := http.Serve(l.cfg.Listener, http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		if l.cfg.HTTPMiddleware.HandleRequest(rw, req) {
-			return
-		}
 
-		// Requests from forward proxy have original hostnames instead of
-		// localhost. Set appropriate header to keep this information.
-		if addr, err := utils.ParseAddr(req.Host); err == nil && !addr.IsLocal() {
-			req.Header.Set("X-Forwarded-Host", req.Host)
-		}
+	server := &http.Server{
+		Handler: http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
Maybe move the handler to be the last field, so we read the timeouts first and there isn't a "strange" indent shift at the end?

Reply:
Done